From e6b6035e22b16da505662362516a2cd3c4af8403 Mon Sep 17 00:00:00 2001
From: Andrzej Surdej <Andrzej_Surdej@comcast.com>
Date: Thu, 15 Dec 2022 15:54:25 +0000
Subject: [PATCH] comcast - DELIA-59087 - Disable pausing playback for
 buffering

1) Disable buffering for progressive playback
2) Set fixed playbin buffer-size so it won't try to estimate
   based on platform dependent bitrate reports
---
 .../graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp b/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp
index 2aec52c4c978..71719b8fe2a8 100644
--- a/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp
+++ b/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp
@@ -2128,6 +2128,9 @@ void MediaPlayerPrivateGStreamer::updateBufferingStatus(GstBufferingMode mode, d
     else
         m_fillTimer.stop();
 
+    // Force buffering flag to be always false so we don't enter buffering state at all
+    m_isBuffering = false;
+
     m_bufferingPercentage = percentage;
     switch (mode) {
     case GST_BUFFERING_STREAM: {
@@ -2962,6 +2965,13 @@ void MediaPlayerPrivateGStreamer::createGSTPlayBin(const URL& url)
         return;
     }
 
+    if (!isMediaSource() && !isMediaStream) { // for progressive playback only
+        // set fixed playbin buffer-size to disable uridecodebin estimations
+        // that could be platform dependent. Use default queue2 elem max size value
+        static constexpr gint kDefaultQueue2BufferSize = 2 * 1024 * 1024;   // Default queue2 size - 2 MB
+        g_object_set(G_OBJECT(pipeline()), "buffer-size", kDefaultQueue2BufferSize, nullptr);
+    }
+
     setStreamVolumeElement(GST_STREAM_VOLUME(m_pipeline.get()));
 
     GST_INFO_OBJECT(pipeline(), "Using legacy playbin element: %s", boolForPrinting(m_isLegacyPlaybin));
-- 
2.25.1

